modules/tensor_mechanics/test/tests/uel/tests

```diff
@@ -0,0 +1,203 @@
+[Tests]
+  issues = '#25163'
+  design = 'AbaqusUserElement.md'
+
+  [build_uel_tri_tests]
+    type = RunCommand
+    command = "cd ../../../examples/uel_tri_tests; make"
+    requirement = "The uel_tri_tests example shall be built."
+    method = 'opt'
+  []
+  [build_uel_tri_states_tests]
+    type = RunCommand
+    command = "cd ../../../examples/uel_tri_states_tests; make"
+    requirement = "The uel_tri_states_tests example shall be built."
+    method = 'opt'
+  []
+  [build_uel_build_tests]
+    type = RunCommand
+    command = "cd ../../../examples/uel_build_tests; make"
+    requirement = "The uel_build_tests example shall be built."
+    method = 'opt'
+  []
+  [build_uel_tri_tests_dbg]
+    type = RunCommand
+    command = "cd ../../../examples/uel_tri_tests; METHOD=dbg make"
+    requirement = "The uel_tri_tests example shall be built."
+    method = 'dbg'
+  []
+  [build_uel_tri_states_tests_dbg]
+    type = RunCommand
+    command = "cd ../../../examples/uel_tri_states_tests; METHOD=dbg make"
+    requirement = "The uel_tri_states_tests example shall be built."
+    method = 'dbg'
+  []
+  [build_uel_build_tests_dbg]
+    type = RunCommand
+    command = "cd ../../../examples/uel_build_tests; METHOD=dbg make"
+    requirement = "The uel_build_tests example shall be built."
+    method = 'dbg'
+  []
+  [reference]
+    type = Exodiff
+    input = reference.i
+    exodiff = reference_out.e
+    abs_zero = 5e-12
+    requirement = "The system shall be able to solve a simple mechanics problem with small elastic "
+                  "deformation with a triangular mesh using MOOSE capabilities."
+    # Only test plugins in dbg and opt.
+    method = 'dbg opt'
+  []
+  [small]
+    type = Exodiff
+    input = small.i
+    exodiff = small_out.e
+    abs_zero = 5e-12
+    requirement = "The system shall be able to solve a simple mechanics problem with small elastic "
+                  "deformation with a triangular mesh using a UEL plugin to that effect that "
+                  "interfaces with the AbaqusUserElement user object."
+    # Only test plugins in dbg and opt.
+    method = 'dbg opt'
+  []
+  [small_test]
+    type = Exodiff
+    input = small_test.i
+    exodiff = small_test_out.e
+    abs_zero = 5e-12
+    requirement = "The system shall be able to solve a simple mechanics problem with small elastic "
+                  "deformation with a triangular mesh using a UEL plugin to that effect that "
+                  "interfaces with the AbaqusUserElement user object and calls a UMAT routine with "
+                  "its standard interface."
+    # Only test plugins in dbg and opt.
+    method = 'dbg opt'
+  []
+  [uel_test_print]
+    type = RunApp
+    input = uel_test_print.i
+    library_mode = 'DYNAMIC'
+    valgrind = 'NONE'
+    # Testing output
+    max_parallel = 1
+    errors = 'StringThatWillNotBePresentInOutput'
+    expect_out = 'Number of predefined variables:            2\n External field number            1  '
```

Comment:
Oh my. This doesn't seem likely to be future proof. Is there some subset out output we can test

Reply:
Why do you think this is not future proof? We control what the output looks like in our test plugin. Unless FORTRAN output formatting changes in the future this will stay the same.

Reply:
Eh this is fine